core/model/src/main/java/org/eclipse/rdf4j/model/IRI.java

```diff
@@ -27,7 +27,7 @@
  * @author Jeen Broekstra
  */
 @SuppressWarnings("deprecation")
-public interface IRI extends URI, Resource {
+public interface IRI extends URI, Resource, Comparable<IRI> {
```

Comment:
We use Comparators for our RDF term types instead of declaring them Comparable, precisely because there is not a single mandated natural ordering for them.

Reply:
To give but one example: the native store keeps track of revisions for IRIs. Either you take that revision into account for total ordering (in which case the comparison is no longer purely lexical) or you don't, in which case comparing different revisions of the same NativeIRI will appear to be equal.

Reply:
We can get away with introducing this, but only if we _very_ explicitly document that we expect all IRI implementations to use lexical ordering for implementing Comparable. This also should make it clear that for some (implementation-specific) types of comparison (e.g. the nativestore one I mentioned above) you should continue to use a bespoke Comparator rather than rely on the Comparable implementation.

Reply:
Regardless, this makes the change backwards incompatible because it is quite conceivable that third party implementors have made their own IRI implementations implement Comparable already.

Reply:
I don't this this change makes it backwards incompatible since there is a default method. And anyone who has made their class also comparable shouldn't really have any issue since it would be a subclass of IRI.